### PR TITLE
Convert more files to Kotlin

### PR DESCRIPTION
**Pull request by ferranpons**

### CODE REVIEW

File: leku/src/main/java/com/schibstedspain/leku/geocoder/AndroidGeocoderDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            try {
+                emitter.onNext(geocoder.getFromLocation(latitude, longitude, MAX_RESULTS))
+                emitter.onComplete()
+            } catch (e: IOException) {
+                emitter.tryOnError(e)
+            }
+        }
+    }
+
+    companion object {
```
Comment (AlbertVilaCalvo) [inline leku/src/main/java/com/schibstedspain/leku/geocoder/AndroidGeocoderDataSource.kt]: we could just put the constant at the top of the and avoid the companion object

https://discuss.kotlinlang.org/t/what-is-the-advantage-of-companion-object-vs-static-keyword/4034/2

File: leku/src/main/java/com/schibstedspain/leku/geocoder/GeocoderPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    private fun getMergedList(geocoderList: List<Address>, placesList: List<Address>): List<Address> {
+        val mergedList = ArrayList<Address>()
+        mergedList.addAll(geocoderList)
+        mergedList.addAll(placesList)
+        return mergedList
+    }
+
+    companion object {
```
Comment (AlbertVilaCalvo) [inline leku/src/main/java/com/schibstedspain/leku/geocoder/GeocoderPresenter.kt]: same here

File: leku/src/main/java/com/schibstedspain/leku/geocoder/GeocoderRepository.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    fun getFromLocation(latLng: LatLng): Observable<List<Address>> {
+        return androidGeocoder.getFromLocation(latLng.latitude, latLng.longitude)
+                .subscribeOn(Schedulers.newThread())
+                .retry(RETRY_COUNT.toLong())
+                .onErrorResumeNext(googleGeocoder.getFromLocation(latLng.latitude, latLng.longitude))
+    }
+
+    companion object {
```
Comment (AlbertVilaCalvo) [inline leku/src/main/java/com/schibstedspain/leku/geocoder/GeocoderRepository.kt]: and here

File: leku/src/main/java/com/schibstedspain/leku/geocoder/GoogleGeocoderDataSource.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                val addresses = addressBuilder.parseResult(result!!)
+                subscriber.onNext(addresses)
+                subscriber.onComplete()
+            } catch (e: JSONException) {
+                subscriber.onError(e)
+            }
+        }
+    }
+
+    companion object {
```
Comment (AlbertVilaCalvo) [inline leku/src/main/java/com/schibstedspain/leku/geocoder/GoogleGeocoderDataSource.kt]: more

File: leku/src/main/java/com/schibstedspain/leku/geocoder/api/NetworkClient.kt [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        val buffer = ByteArray(maxLength)
+        var length = stream.read(buffer)
+        while (length != -1) {
+            result.write(buffer, 0, length)
+            length = stream.read(buffer)
+        }
+        return result.toString("UTF-8")
+    }
+
+    companion object {
```
Comment (AlbertVilaCalvo) [inline leku/src/main/java/com/schibstedspain/leku/geocoder/api/NetworkClient.kt]: ...

File: leku/src/main/java/com/schibstedspain/leku/geocoder/AndroidGeocoderDataSource.kt (hunk 1 above)
Comment (ferranpons) [reply leku/src/main/java/com/schibstedspain/leku/geocoder/AndroidGeocoderDataSource.kt]: Thanks for the suggestion. I've changed all the constants.

**Pull request merged**